eidas-saml-parser/src/main/java/uk/gov/ida/notification/eidassaml/logging/EidasAuthnRequestAttributesLogger.java

```diff
@@ -0,0 +1,24 @@
+package uk.gov.ida.notification.eidassaml.logging;
+
+import org.opensaml.saml.saml2.core.AuthnRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+public class EidasAuthnRequestAttributesLogger {
+    private static final Logger log = LoggerFactory.getLogger(EidasAuthnRequestAttributesLogger.class);
+
+    public static void logAuthnRequestAttributes(AuthnRequest authnRequest) {
+        try {
+            MDC.put("eidasRequestId", authnRequest.getID());
+            MDC.put("eidasDestination", authnRequest.getDestination());
+            MDC.put("eidasIssueInstant", authnRequest.getIssueInstant().toString());
+            MDC.put("eidasIssuer", authnRequest.getIssuer().getValue());
+            log.info("Authn request validated by ESP");
+        } catch (Exception e) {
+            MDC.clear();
+            throw e;
+        }
```

Comment:
you may be able to use a finally cause for the `MDC.clear()` to save it being repeated.

Reply:
Interesting, I was thinking if there was a better way to do this? How would it look? If an exception is thrown, I do actually want it to throw and interrupt the runtime, I just need to clear the MDC first. If have the `MDS.clear()` in a finally block, won't that let the flow of execution continue?

Reply:
I think with a finally it does still throw, but it throws then tidies up. 

so it be a bit like,

```
        try {
            MDC.put("eidasRequestId", authnRequest.getID());
            MDC.put("eidasDestination", authnRequest.getDestination());
            MDC.put("eidasIssueInstant", authnRequest.getIssueInstant().toString());
            MDC.put("eidasIssuer", authnRequest.getIssuer().getValue());
            log.info("Authn request validated by ESP");
        } catch (Exception e) {
            throw e;
        } finally {
            MDC.clear();
        }

```
https://stackoverflow.com/questions/4264874/does-a-finally-block-run-even-if-you-throw-a-new-exception seems to give more detail.

Reply:
Nice, I'll update to that.